homeassistant/components/lyric/__init__.py

```diff
@@ -76,6 +77,12 @@ async def async_update_data(force_refresh_token: bool = False) -> Lyric:
         try:
             async with asyncio.timeout(60):
                 await lyric.get_locations()
+                for location in lyric.locations:
+                    for device in location.devices:
+                        if device.deviceClass == "Thermostat":
+                            await lyric.get_thermostat_rooms(
```

Comment:
Maybe it would be more performant to await all promises altogether?

https://docs.python.org/3/library/asyncio-task.html#asyncio.gather